src/server/vnparse.c

```diff
@@ -2505,28 +2641,50 @@ add_to_resc_limit_list_sorted(pbs_list_head *phead, resc_limit_t *resc)
 	pbs_list_link	*plink_cur;
 	rl_entry	*p_entry_cur = NULL;
 	rl_entry	*new_resc = NULL;
+	resc_limit_t	*p_res_cur;
 
 	if ((phead == NULL) || (resc == NULL))
 		return (1);
 
-	plink_cur = phead;
-	p_entry_cur = (rl_entry *)GET_NEXT(*phead);
-
-	while (p_entry_cur) {
+	for (plink_cur = phead,
+		p_entry_cur = (rl_entry *)GET_NEXT(*phead);
+		p_entry_cur;
+		p_entry_cur = (rl_entry *)GET_NEXT(*plink_cur)) {
 		plink_cur = &p_entry_cur->rl_link;
+		p_res_cur = p_entry_cur->resc;
 
-		if (p_entry_cur->resc != NULL) {
+		if (p_res_cur != NULL) {
+#if defined(PBS_MOM) || defined(PBS_PYTHON)
 			/* order according to increasing # of cpus
 			 * if same # of cpus, use increasing amt of mem
 			 */
-			if ((p_entry_cur->resc->rl_ncpus > resc->rl_ncpus) ||
-			      ((p_entry_cur->resc->rl_ncpus == resc->rl_ncpus) &&
-				  (p_entry_cur->resc->rl_mem > resc->rl_mem))) {
+			if ((p_res_cur->rl_ncpus > resc->rl_ncpus) ||
+			      ((p_res_cur->rl_ncpus == resc->rl_ncpus) &&
+				  (p_res_cur->rl_mem > resc->rl_mem))) {
 				break;
 			}
-		}
+#else
+			if (p_res_cur->rl_res_count < resc->rl_res_count)
+				continue;
+			else if (p_res_cur->rl_res_count > resc->rl_res_count)
+				break;
+			else {
```

Comment:
This else block doesn't need to be here since this is the only code path we can take if we didn't break or continue. Don't get rid of the block, just don't indent it.

Reply:
handled in commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a